packages/babel-parser/src/parser/statement.js

```diff
@@ -2091,7 +2071,7 @@ export default class StatementParser extends ExpressionParser {
 
   shouldParseExportDeclaration(): boolean {
     const { type } = this.state;
-    if (type === tt.at) {
+    if (type === tt.at || type === tt.atInit) {
```

Comment:
We don't need to expect `decorators-legacy` plugin when `type` is `tt.atInit`, since the token is invalid in the legacy decorators.